test/lib/server/services/user/UserService.test.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const { expect } = require('chai');
+const { userService } = require('../../../../../lib/server/services/user/UserService.js');
+const { getUser } = require('../../../../../lib/server/services/user/getUser.js');
+const { createOrUpdateUser } = require('../../../../../lib/server/services/user/createOrUpdateUser.js');
+
+module.exports = () => {
+    it('should successfully get the existing user with getOrCreate function', async () => {
+        const externalId = 1001;
+        const name = 'testUser';
+
+        // Create user in db
+        await createOrUpdateUser({ personid: externalId, name });
+
+        const user = await userService.getOrCreate({ externalUserId: externalId }, name);
+
+        // User should exist now
+        expect(user).not.to.be.null;
+        // Id should be filled automatically
+        expect(user.id).not.to.be.null;
+        // ExternalId and name should be equal
+        expect(user.externalId).to.equal(externalId);
+        expect(user.name).to.equal(name);
```

Comment:
Here this seems to me to be a bit of cyclic testing. Don't you have seeders for users?
If you don't do not bother it's still ok

Reply:
I believe I do not have seeders. I try to make the tests so that they create data independently, so that they can run individually, like you adviced me to do because some tests are dependent on each other and that makes creating new tests a bit of a struggle sometimes, for example, for runs.
So, that is the main reason why I create the data in the same test.